src/layouts/MainLayout/MainLayout.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <div class="mainLayout">
+    <div>
+      <Suspense>
```

Comment:
Maybe add a `fallback` to this suspense, please check https://vuejs.org/guide/built-ins/suspense.html#loading-state
```suggestion
      <Suspense>
        <router-view />
        <template #fallback>
          Loading...
        </template>
      </Suspense>
```

Reply:
Thanks for the suggestion, I will check it ! 😁

Reply:
Great! I have tested it and it's a good way to show a loading page:

![LastRecord](https://user-images.githubusercontent.com/66505715/172071011-4c367150-4798-42e3-91ac-4ef032335d51.gif)

It will require some changes on each component but I think it's not problem

Reply:
Maybe the refactor could be done on another PR

Reply:
Although the loading is showing as expected, I'm receiving some errors from async setup components, i will check why

![image](https://user-images.githubusercontent.com/66505715/172072242-552ced2e-a99f-4904-9ad7-01bb0644ff57.png)

You can test it if you enter on https://deploy-preview-40--did-app.netlify.app/profile/

Reply:
I understand the error!✌️.

Vue Router is searching the "router-view" component, which shows the view (the child) on the template.

Suspense is using a type of conditional to show "Loading..." while the component end of loads.

Then as Vue Router hasn't found the component <router-view /> it threw the error.

I've solved it using the v-slot of router-view to get the component given for Vue Router, then using Suspense to show the Loading if is needed, or the Component if it's already loaded

![code](https://user-images.githubusercontent.com/66505715/172072565-d7816af1-8de5-4e09-8f65-01afa3a70727.png)

Reply:
Good work man!

Reply:
https://github.com/Azordev/did-app/issues/41 I've created an issue to refactor all components and add loading component